src/app/zap-templates/zcl/data-model/chip/time-synchronization-cluster.xml

```diff
@@ -0,0 +1,87 @@
+<?xml version="1.0"?>
+<!--
+Copyright (c) 2021 Project CHIP Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+<configurator>
+  <domain name="CHIP"/>
+
+  <struct name="TimeZoneType">
+    <item name="Offset" type="INT32S" min="-43200" max="50400"/>
+    <item name="ValidAt" type="INT64U"/> 
+    <item name="Name" type="CHAR_STRING" length="64"/>
+  </struct>
+
+  <struct name="DstOffsetType">
+    <item name="Offset" type="INT32S"/>
+    <item name="ValidStarting" type="INT64U"/> 
+    <item name="ValidUntil" type="INT64U"/>
+  </struct>
+
+
+  <enum name="Granularity" type="ENUM8">
+    <item name="NoTimeGranularity" value="0x0"/>
+    <item name="MinutesGranularity" value="0x1"/>
+    <item name="SecondsGranularity" value="0x2"/>
+    <item name="MillisecondsGranularity" value="0x3"/>
+    <item name="MicrosecondsGranularity" value="0x4"/>
+  </enum>
+
+  <enum name="TimeSourceEnum" type="ENUM8">
+    <item name="None" value="0x0"/>
+    <item name="Unknown" value="0x1"/>
+    <item name="Admin" value="0x2"/>
+    <item name="NodeTimeCluster" value="0x3"/>
+    <item name="NonFabricSntp" value="0x4"/>
+    <item name="NonFabricNtp" value="0x5"/>
+    <item name="FabricSntp" value="0x6"/>
+    <item name="FabricNtp" value="0x7"/>
+    <item name="MixedNtp" value="0x8"/>
+    <item name="NonFabricSntpNts" value="0x9"/>
+    <item name="NonFabricNtpNts" value="0xa"/>
+    <item name="FabricSntpNts" value="0xb"/>
+    <item name="FabricNtpNts" value="0xc"/>
+    <item name="MixedNtpNts" value="0xd"/>
+    <item name="CloudSource" value="0xe"/>
+    <item name="Ptp" value="0xf"/>
+    <item name="Gnss" value="0x10"/>
+  </enum>
+
+  <cluster>
+    <domain>General</domain>
+    <name>TimeSync</name>
+    <code>0x0038</code>
+    <define>TIMESYNC_CLUSTER</define>
+    <description>This cluster is used for time synchronization.</description>
+
+    <attribute side="server" code="0x0001" define="UTC_TIME" type="INT64U" writable="false" default="" optional="false">UTCTime</attribute>
```

Comment:
Does `type="epoch_us"` not work?

Reply:
It doesn't seem to work. The existing clusters such as `basic-information-cluster.xml` also don't use custom basic types (e.g. `int16u` instead of `vendor_id`). Using custom basic types causes the ZCL code generation to fail.

Reply:
@vivien-apple @tecimovic Is there an issue tracking making this sort of thing work?